### PR TITLE
Add the remasters as install source for the old(!) assets

### PR DESCRIPTION
**Pull request by abcdefg30**
I finished RA first. The remasters don't ship the soviet videos so we can't install those.

### CODE REVIEW

File: mods/ra/installer/remaster.yaml [hunk 1]
```diff
@@ -0,0 +1,329 @@
+ra-remaster-steam: C&C Remastered (Steam version, English)
+	Type: Install
+	RegistryPrefixes: HKEY_LOCAL_MACHINE\SOFTWARE\
+	RegistryKey: Microsoft\Windows\CurrentVersion\Uninstall\Steam App 1213210
```
Comment (pchote) [inline mods/ra/installer/remaster.yaml]: I'd prefer if we could expose this as a distinct `SteamAppID` key, with new install logic that knows to look at this registry key on windows or do whatever is needed to find the path into the proton container on Linux.

This feature will also be used by the KKnD mod.

File: mods/ra/installer/remaster.yaml [hunk 2]
```diff
@@ -0,0 +1,329 @@
+ra-remaster-steam: C&C Remastered (Steam version, English)
```
Comment (pchote) [inline mods/ra/installer/remaster.yaml:1]: We should aim to support Origin too before we ship a release with this.

**Comment by abcdefg30** [comment]
Updated and added support for TD as well. I don't know how to set Origin up however.

**Comment by pchote** [comment]
Finding the steam install on macOS (and I assume Linux) is a little more complicated.

The primary steam library location is found at `~/Library/Application Support/Steam/steamapps`. Additional steam library locations are listed in the `libraryfolders.vdf` file in the root of the primary library:
```
"LibraryFolders"
{
	"TimeNextStatsReport"		"<removed>"
	"ContentStatsID"		"<removed>"
	"1"		"/path/to/another/SteamLibrary"
}
```
Installed games write an `appmanifest_<appid>.acf` file to the root of whatever library they are installed to:
```
"AppState"
{
	"appid"		"427520"
	"Universe"		"1"
	"name"		"Factorio"
	"StateFlags"		"4"
	"installdir"		"Factorio"
	"LastUpdated"		"<removed>"
	"UpdateResult"		"0"
	"SizeOnDisk"		"1521936185"
	"buildid"		"4404625"
	"LastOwner"		"<removed>"
	"BytesToDownload"		"588319168"
	"BytesDownloaded"		"588319168"
	"AutoUpdateBehavior"		"0"
	"AllowOtherDownloadsWhileRunning"		"0"
	"ScheduledAutoUpdate"		"0"
	"InstalledDepots"
	{
		"427525"
		{
			"manifest"		"4837874827908946167"
		}
	}
	"MountedDepots"
	{
		"427525"		"4837874827908946167"
	}
	"UserConfig"
	{
		"language"		"english"
	}
}
```

The `installdir` key gives the directory name within the library's `common` directory which holds the (os specific!) game files. For example, Factorio on macOS has a `Factorio.app` which obviously won't be the case on Windows or Linux.

I assume that games installed via Proton (which will be the case for the remaster on Linux) will have additional keys in the appmanifest and a different layout inside the install directory. I will test this on one of my Linux vms when I have a chance.

**Comment by pchote** [comment]
It looks like the same is true on Windows, so it would IMO be best if we could use the same logic on all platforms instead of special-casing Windows to look in the registry.

**Comment by abcdefg30** [comment]
We have another issue: I discovered that some of the `aud` files are mono not stereo, which we can't play right now.

**Comment by pchote** [comment]
I expect we can agree that various things have increased the scope larger than we reasonably want to tackle in this release cycle, so bumping to Next + 1.

**Comment by reaperrr** [comment]
> We have another issue: I discovered that some of the aud files are mono not stereo, which we can't play right now.

Uh, all original `aud` files were mono, though? Or am I misreading what you mean?

**Comment by abcdefg30** [comment]
I was referring to this: https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Cnc/FileFormats/AudReader.cs#L23
XCC Mixer indeed reports all of them as "mono", but the ones we can play are 16 bit while the remasters have 8 bit ones. (Which afaik meant the 8 bit version had a field set to 0 while all files we use have the same field with value 1, which is then parsed as the correct `SoundFlag`, while 0 has no corresponding flag.)

**Comment by abcdefg30** [comment]
We still haven't fixed the audio and this might not be too useful if we go the direct steam-detection route, so closing for now.

**Pull request closed without merging**